src/asyncProfiler.cpp

```diff
@@ -98,9 +97,24 @@ void Profiler::storeCallTrace(ASGCT_CallTrace* trace) {
         }
         if (++i == MAX_CALLTRACES) i = 0;
     } while (i != bucket);
+    
+    if (_frameBufferSize - _freeFrame < trace->num_frames) {
+        // Don't have enough space to store call trace
+        _frameBufferOverflow = true;
+        return;
+    }
```

Comment:
BTW, we might not be able to increment the counter for a huge trace, which is **already** stored, but there is not enough space for another instance of this trace.